lib/rets/http_client.rb

```diff
@@ -81,11 +81,12 @@ def rets_extra_headers
       headers
     end
 
-    def http_cookie(name)
-      http.cookies.each do |c|
-        return c.value if c.name.downcase == name.downcase && c.match?(URI.parse(login_url))
-      end
-      nil
+    def http_cookie
+      # Use CookieManager#cookie_value to obtain cookie value.
+      # This is independent of cookie implementation, i.e. CookieManager can use
+      # Webagent::Cookie or Http::Cookie underneath
+      cookie_values = http.cookie_manager.cookie_value URI.parse(login_url)
+      cookie_values.split("; ", 1).first.split("=").last if cookie_values
```

Comment:
I'm wary of adding so much knowledge of cookie formatting as well as making the assumption there is only one cookie. I like the idea of the cookie manager, but would rather depend on something else to do the parsing, so that the original code, or something similar to it can still work.

I haven't dug into this code enough to know the right way to do it... @hfaulds Do you mind taking a look at this?